src/spdx2/agent/spdx2.php

```diff
@@ -716,7 +717,10 @@ protected function generateFileNodesByFiles($filesWithLicenses, $treeTableName)
       {
         $licenses['scanner'] = array();
       }
-      $content .= $this->renderString($this->getTemplateFile('file'),array(
+      if(!$this->getIgnoreFilesWOinfo($uploadId) ||
+         ($this->getIgnoreFilesWOinfo($uploadId) && (!empty($licenses['concluded']) || (!empty($licenses['scanner']) && !empty($licenses['scanner'][0])) || !empty($licenses['copyrights']))))
```

Comment:
I guess we could leave out the second call to `$this->getIgnoreFilesWOinfo($uploadId) ` and retain the same logic.
This would save an unnecessary call DB query.


```
      if(!$this->getIgnoreFilesWOinfo($uploadId) ||
         (!empty($licenses['concluded']) || (!empty($licenses['scanner']) && !empty($licenses['scanner'][0])) || !empty($licenses['copyrights'])))
```